### PR TITLE
Authorization

### PR DESCRIPTION
**Pull request by alistkov**

### CODE REVIEW

**Review by dmitriimokienko — APPROVED** [review]

File: src/authentication/authentication.service.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    private readonly jwtService: JwtService,
+    @Inject(jwtConfig.KEY)
+    private readonly jwtConfiguration: ConfigType<typeof jwtConfig>,
+  ) {}
+
+  async signUp(signUpDto: SignUpDto) {
+    try {
+      const user = new User();
+      user.email = signUpDto.email;
+      user.password = await this.hashingService.hash(signUpDto.password);
```
[the remark this answers is not in the document]
Comment (alistkov) [reply src/authentication/authentication.service.ts:31]: now class-validator throws exceptions if field fails validation
**Example:**
```
Password field

@Length(8, 16)
password: string;
```
```
Validation error

{
    "message": [
        "password must be longer than or equal to 8 characters"
    ],
    "error": "Bad Request",
    "statusCode": 400
}
```

[the remark this answers is not in the document]
Comment (dmitriimokienko) [reply src/authentication/authentication.service.ts:31]: password has more special rules, for example: special symbols, required Low & upper case letters, require digits

I'm not sure that we can add this rules via decorators, if we can, ok

**Review by dmitriimokienko — APPROVED** [review]

File: src/authentication/authentication.service.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (dmitriimokienko) [reply src/authentication/authentication.service.ts:31]: I found solution

https://stackoverflow.com/questions/60451337/password-confirmation-in-typescript-with-class-validator

also lib has `@isStrongPassword`

[the remark this answers is not in the document]
Comment (alistkov) [reply src/authentication/authentication.service.ts:31]: Yep. I found this solution too. I am checking the docs and will pick better solution for this.

[the remark this answers is not in the document]
Comment (alistkov) [reply src/authentication/authentication.service.ts:31]: Add `@Matches` decorator to validate password to match regular expression

**Review by dmitriimokienko — APPROVED** [review]

**Pull request merged**